dora/stress/shell/src/main/java/alluxio/stress/cli/worker/StressWorkerBench.java

```diff
@@ -455,35 +456,71 @@ private void runInternal() throws Exception {
       CommonUtils.sleepMs(waitMs);
       SAMPLING_LOG.info("Test started and recording will be started after the warm up at {}",
           CommonUtils.convertMsToDate(recordMs, dateFormat));
+
+      String workerID = mBaseParameters.mIndex;
+      int lastDashIndex = workerID.lastIndexOf("-");
+      if (lastDashIndex != -1) {
+        workerID = toString().substring(lastDashIndex + 1);
+      }
+      WorkerBenchCoarseDataPoint dp = new WorkerBenchCoarseDataPoint(
+          Long.parseLong(workerID),
+          Thread.currentThread().getId(),
+          new ArrayList<>(), new ArrayList<>());
+      List<WorkerBenchDataPoint> dpList = new ArrayList<>();
+      List<Long> throughputList = new ArrayList<>();
+      int lastSlice = 0;
+
       while (!Thread.currentThread().isInterrupted()
           && CommonUtils.getCurrentMs() < mContext.getEndMs()) {
         // Keep reading the same file
-        WorkerBenchDataPoint dataPoint = applyOperation();
-        long currentMs = CommonUtils.getCurrentMs();
-        // Start recording after the warmup
-        if (currentMs > recordMs) {
-          mResult.addDataPoint(dataPoint);
-          if (dataPoint.getIOBytes() > 0) {
-            mResult.incrementIOBytes(dataPoint.getIOBytes());
+        long startMs = CommonUtils.getCurrentMs() - recordMs;
+        long bytesRead = applyOperation();
+        long duration = CommonUtils.getCurrentMs() - recordMs - startMs;
+        if (startMs > 0) {
+          if (bytesRead > 0) {
+            mResult.setIOBytes(mResult.getIOBytes() + bytesRead);
+            throughputList.add(bytesRead / duration);
+            int currentSlice = (int) (startMs
+                / FormatUtils.parseTimeSize(mParameters.mSliceLength));
+            while (currentSlice > lastSlice) {
```

Comment:
Yes it will. For example, if the slice size is 100ms, and if there are reads >200ms, there might be slices without a single record. The slice ID is the index of the array, so we need to record empty slices as well.